taraxa/state/contracts/dpos/precompiled/validators.go

```diff
@@ -139,43 +156,66 @@ func (self *Validators) CreateValidator(owner_address *common.Address, validator
 	rewards.CommissionRewardsPool = big.NewInt(0)
 	Save(self, validator_address, rewards)
 
-	validator_owner_key := stor_k_1(self.validator_owner_field, validator_address[:])
+	validator_owner_key := contract_storage.Stor_k_1(self.validator_owner_field, validator_address[:])
 	self.storage.Put(validator_owner_key, owner_address.Bytes())
 
-	validator_vrf_key := stor_k_1(self.validator_vrf_key_field, validator_address[:])
+	validator_vrf_key := contract_storage.Stor_k_1(self.validator_vrf_key_field, validator_address[:])
 	self.storage.Put(validator_vrf_key, vrf_key)
 
 	// Adds validator into the list of all validators
 	self.validators_list.CreateAccount(validator_address)
-	return validator
+
+	return
 }
 
 func (self *Validators) DeleteValidator(validator_address *common.Address) {
-	validator_key := stor_k_1(self.validator_field, validator_address[:])
+	validator_key := contract_storage.Stor_k_1(self.validator_field, validator_address[:])
 	self.storage.Put(validator_key, nil)
 
-	validator_info_key := stor_k_1(self.validator_info_field, validator_address[:])
+	validator_info_key := contract_storage.Stor_k_1(self.validator_info_field, validator_address[:])
 	self.storage.Put(validator_info_key, nil)
 
-	validator_owner_key := stor_k_1(self.validator_owner_field, validator_address[:])
+	validator_owner_key := contract_storage.Stor_k_1(self.validator_owner_field, validator_address[:])
 	self.storage.Put(validator_owner_key, nil)
 
-	validator_vrf_key := stor_k_1(self.validator_vrf_key_field, validator_address[:])
+	validator_vrf_key := contract_storage.Stor_k_1(self.validator_vrf_key_field, validator_address[:])
 	self.storage.Put(validator_vrf_key, nil)
 
-	rewards_key := stor_k_1(self.validator_rewards_field, validator_address[:])
+	rewards_key := contract_storage.Stor_k_1(self.validator_rewards_field, validator_address[:])
 	self.storage.Put(rewards_key, nil)
 
 	// Removes validator from the list of all validators
 	self.validators_list.RemoveAccount(validator_address)
 }
 
 func (self *Validators) GetValidator(validator_address *common.Address) (validator *Validator) {
-	return Get[Validator](self, validator_address)
+	key := contract_storage.Stor_k_1(self.validator_field, validator_address[:])
+	self.storage.Get(key, func(bytes []byte) {
+		// Try to decode into post-hardfork extented Validator struct first
+		validator = new(Validator)
+		validator.ValidatorV1 = new(ValidatorV1)
+
+		err := rlp.DecodeBytes(bytes, validator)
+		if err != nil {
+			// Try to decode into pre-hardfork ValidatorV1 struct
+			err = rlp.DecodeBytes(bytes, validator.ValidatorV1)
+			validator.UndelegationsCount = 0
+			if err != nil {
+				// This should never happen
+				panic("Unable to decode validator rlp")
+			}
+		}
+	})
```

Comment:
Looks identical to what we have in the reader. Maybe we can move it somewhere to avoid code duplication